pkg/routeagent/controllers/route/route.go

```diff
@@ -310,6 +326,43 @@ func (r *Controller) updateIptableRulesForInterclusterTraffic(inputCidrBlocks []
 	}
 }
 
+func (r *Controller) updateRoutingRulesForHostNetworkSupport(inputCidrBlocks []string, operation Operation) {
+	if operation == FlushRouteTable {
+		r.routeCacheGWNode.DeleteAll()
+		cmd := exec.Command("/sbin/ip", "r", "flush", "table", strconv.Itoa(RouteAgentHostNetworkTableID))
+		if err := cmd.Run(); err != nil {
+			// We can safely ignore this error, as this table
+			// won't exist in most nodes (only gateway nodes)
+			return
+		}
+	} else {
+		// These routing rules are required ONLY on the Gateway Node.
+		// On the non-Gateway nodes, we use iptable rules to support this use-case.
+		if r.isGatewayNode && r.cniIface != nil {
```

Comment:
This can be combined with L338 to reduce indentation.

Reply:
Done